components/UserTestimonies/Tabs.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {
+  MouseEventHandler,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState
+} from "react"
+import styled from "styled-components"
+
+type onClickEventFunction = (e: Event, value: number) => void
+
+export const TabSlider = (props: { width: number; position: number }) => {
+  const { width, position } = props
+  return <TabSliderStyle width={width ?? 200} position={position} />
+}
+
+export const TabSliderContainer = (props: { children?: JSX.Element }) => {
+  const { children } = props
+  return <TabSliderContainerStyle>{props.children}</TabSliderContainerStyle>
+}
+
+export const Tab = (props: {
+  label: string
+  value: number
+  active: boolean
+  onClick?: MouseEventHandler
+}) => {
+  const { label, onClick, active } = props
+  return (
+    <TabStyle onClick={onClick} active={active}>
+      <h3> {label}</h3>
+    </TabStyle>
+  )
+}
+
+export const Tabs = (props: {
+  childTabs: ReactElement[]
+  onChange: onClickEventFunction
+  selectedTab: number
+}) => {
+  const containerRef = useRef(null)
+  const tabRefs = useRef<Array<HTMLDivElement | null>>([])
+  const { childTabs, onChange, selectedTab } = props
+  const [sliderWidth, setSliderWidth] = useState(0)
+  const [sliderPos, setSliderPos] = useState(0)
+  const sliderPositionOffset = 16
+
+  useEffect(() => {
+    if (
+      tabRefs.current[selectedTab - 1] !== null &&
+      tabRefs.current[selectedTab - 1]!.getBoundingClientRect().width
+    ) {
+      setSliderWidth(
+        tabRefs.current[selectedTab - 1]!.getBoundingClientRect().width
+      )
```

Comment:
To handle resizing, I like to use the [`useResizeObserver` hook from `beautiful-react-hooks`](https://antonioru.github.io/beautiful-react-hooks/#/useResizeObserver) (you'll have to add the dependency).

You would attach a ref to the TabSliderContainer, get its width with the hook, add its width as a dependency to this useEffect, so we update the width/position of the tab whenever the container changes size.